pulp_2_tests/tests/rpm/api_v2/test_modular_errata.py

```diff
@@ -139,13 +150,102 @@ def test_collection_field(self):
             'collection names not proper'
         )
 
+    def test_copy_errata(self):
+        """Test whether Errata modules are copied.
+
+        This Test does the following:
+
+        1. It creates, syncs, and publishes a modules rpm repository.
+        2. Creates another repo with no feed.
+        3. Recursively copies an errata from one repo to another.
+        4. Checks whether the errata information in the new repo is
+           correct.
+        """
+        repo_1, _ = self._set_repo_and_get_repo_data()
+
+        # Creating an empty repo2
+        body = gen_repo(distributors=[gen_distributor(auto_publish=True)])
+        repo_2 = self.client.post(REPOSITORY_PATH, body)
+        self.addCleanup(self.client.delete, repo_2['_href'])
+
+        criteria = {
+            'filters': {
+                'unit': {
+                    'id': MODULE_FIXTURES_ERRATA['errata_id']
+                }},
+            'type_ids': ['erratum']
+        }
+
+        # Copy errata data recursively from repo1 to repo2
+        self.client.post(urljoin(repo_2['_href'], 'actions/associate/'), {
+            'source_repo_id': repo_1['id'],
+            'override_config': {'recursive': True},
+            'criteria': criteria
+        })
+        repo_2 = self.client.get(repo_2['_href'], params={'details': True})
+
+        self.assertEqual(
+            repo_2['total_repository_units'],
+            MODULE_FIXTURES_ERRATA['total_available_units'],
+            repo_2
+        )
+
+        self.assertEqual(
+            search_units(self.cfg, repo_1, criteria)[0]['metadata']['pkglist'],
+            search_units(self.cfg, repo_2, criteria)[0]['metadata']['pkglist'],
+            'Copied erratum doesn''t contain the same module/rpms'
+        )
+
+    def test_upload_errata(self):
+        """Upload errata and check whether it got published in the repo.
+
+        This Test does the following.
+        1. Create, sync, publish a repo with RPM_Unsigned_feed_url.
+        2. Upload a custom modular erratum to the repo. The custom
+           module erratum is obtained from ``_get_erratum()``.
+        3. Verify whether the uploaded erratum is present in the repo.
+        """
+        # Step 1
+        body = gen_repo(
+            importer_config={'feed': RPM_UNSIGNED_FEED_URL},
+            distributors=[gen_distributor(auto_publish=True)]
```

Comment:
Evaluate just publish the repo after the errata is upload as described in #94.

Reply:
@ragabala, I confirmed with @dparalen, that the repo was synced, published then the errata was uploaded.  It is fine to keep the `auto_publish` as True.